src/evict/evict_lru.c

```diff
@@ -2150,15 +2150,23 @@ __evict_walk_prepare(WT_SESSION_IMPL *session, uint32_t *walk_flagsp)
     switch (btree->evict_start_type) {
     case WT_EVICT_WALK_NEXT:
         /* Each time when evict_ref is null, alternate between linear and random walk */
-        if (btree->evict_ref == NULL && (++btree->linear_walk_restarts) & 1)
-            /* Alternate with rand_prev so that the start of the tree is visited more often */
-            goto rand_prev;
+        if (btree->evict_ref == NULL && (++btree->linear_walk_restarts) & 1) {
+            if (S2C(session)->evict->use_npos_in_pass)
+                /* Alternate with rand_prev so that the start of the tree is visited more often */
+                goto rand_next;
+            else
+                goto rand_prev;
```

Comment:
Looks like it used to be rand_next [before](https://github.com/wiredtiger/wiredtiger/commit/f552cf9638735ab8907bf5d1a21e3149a3c13902#diff-fba173dbad7997f847117319e1939715f1969181ab2937508a2c9303b201f017L2020).

Reply:
done